Add Privacy Policy consent checkbox

ilicfilip opened the pull request:
## Context
Add Privacy Policy consent checkbox to the Onboard form.

## Quality assurance

* [x] I have tested this code to the best of my abilities.
* [ ] I have added unit tests to verify the code works as intended.
* [x] I have checked that the base branch is correctly set.

aristath commented:
@ilicfilip 

* I deleted the `progress_planner_license_key` option by adding `\delete_option( 'progress_planner_license_key' );` in the main plugin file
* Went to the plugin page
* Got the popover
* Selected NOT to receive the emails
* Clicked to accept the terms and submitted the form

I got the message `Success! We saved your data on ubuntu.orb.local so we can email you every week.` - I'm not sure that is related to this PR, but the message should not show up if the user has not opted-in to emails.

ilicfilip commented:
Weird, I tried that many times in last couple of days - debugging.

ilicfilip commented:
Fixed, welcome popup was removed too fast for me so that is the reason why I missed it.

I have added a new message in case user account is not created, possibly wording needs to be adjusted.
